src/plugin/plugin.ts

```diff
@@ -865,12 +1166,53 @@ class Purchases {
       [
         aPackage.identifier,
         aPackage.offeringIdentifier,
-        upgradeInfo !== undefined && upgradeInfo !== null
-          ? upgradeInfo.oldSKU
-          : null,
-        upgradeInfo !== undefined && upgradeInfo !== null
-          ? upgradeInfo.prorationMode
+        oldProductIdentifier,
+        prorationMode,
+        googleIsPersonalizedPrice,
+      ]
+    );
+  }
+
+  /**
+   * Google only. Make a purchase of a subscriptionOption
+   *
+   * @param {SubscriptionOption} subscriptionOption The SubscriptionOption you wish to purchase. You can get the SubscriptionOption from StoreProducts by calling getOfferings
+   * @param {function(string, CustomerInfo):void} callback Callback triggered after a successful purchase.
+   * @param {function(PurchasesError, boolean):void} errorCallback Callback triggered after an error or when the user cancels the purchase.
+   * If user cancelled, userCancelled will be true
+   * @param {GoogleProductChangeInfo} googleProductChangeInfo Android only. Optional GoogleProductChangeInfo you 
+   * wish to upgrade from containing the oldProductIdentifier and the optional prorationMode.
+   * @param {boolean} googleIsPersonalizedPrice Android and Google only. Optional boolean indicates personalized pricing on products available for purchase in the EU.
+   * For compliance with EU regulations. User will see "This price has been customize for you" in the purchase dialog when true.
+   * See https://developer.android.com/google/play/billing/integrate#personalized-price for more info.
+   */
+  public static purchaseSubscriptionOption(
+    subscriptionOption: SubscriptionOption,
+    callback: ({productIdentifier, customerInfo,}: { productIdentifier: string; customerInfo: CustomerInfo; }) => void,
+    errorCallback: ({error, userCancelled,}: { error: PurchasesError; userCancelled: boolean; }) => void,
+    googleProductChangeInfo?: GoogleProductChangeInfo | null,
+    googleIsPersonalizedPrice: boolean = false
+  ) {
+    window.cordova.exec(
+      callback,
+      (response: { [key: string]: any }) => {
+        const {userCancelled, ...error} = response;
+        errorCallback({
+          error: error as PurchasesError,
+          userCancelled,
+        });
+      },
+      PLUGIN_NAME,
+      "purchaseSubscriptionOption",
```

Comment:
Hmm do we need to handle this being called in the iOS side? Ideally it just errors out.

Reply:
Maybe?! It requires a `SubscriptionOption` object which is only obtainable through `storeProduct.subscriptionOptions` or `storeProduct.defaultOption` but... should probably still handle gracefully either in iOS or do a platform check in the Cordova level method to prevent iOS from being called 🤷‍♂️

Reply:
Did we end up doing this? I think it would still call iOS code, not sure the behavior it would have there.

Reply:
Just added an unsupported error if it gets called on iOS! The user would have to construct their own `SubscriptionOption` to make this work but it will be handled just in case 🤷‍♂️